### PR TITLE
Alerting: update DeleteAlertRuleByUID to accept many UID

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What this PR does / why we need it**:
This PR updates method DeleteAlertRuleByUID to accept many UID. This will be used in many places in the future.

### CODE REVIEW

File: pkg/services/ngalert/schedule/schedule_test.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ import (
 	"time"
 
 	"github.com/grafana/grafana-plugin-sdk-go/data"
+
```
Comment (joeblubaugh) [inline pkg/services/ngalert/schedule/schedule_test.go:13]: I've seen a lot of variations of import grouping in the grafana codebase. Is there a preferred style - keepiong non-grafana, non-stdlib packages in their own block?

File: pkg/services/ngalert/store/alert_rule.go [hunk 2]
```diff
@@ -64,20 +64,30 @@ func getAlertRuleByUID(sess *sqlstore.DBSession, alertRuleUID string, orgID int6
 }
 
 // DeleteAlertRuleByUID is a handler for deleting an alert rule.
-func (st DBstore) DeleteAlertRuleByUID(ctx context.Context, orgID int64, ruleUID string) error {
+func (st DBstore) DeleteAlertRulesByUID(ctx context.Context, orgID int64, ruleUID ...string) error {
 	return st.SQLStore.WithTransactionalDbSession(ctx, func(sess *sqlstore.DBSession) error {
-		_, err := sess.Exec("DELETE FROM alert_rule WHERE org_id = ? AND uid = ?", orgID, ruleUID)
+		params := "(" + strings.Repeat(",?", len(ruleUID))[1:] + ")"
```
Comment (joeblubaugh) [inline pkg/services/ngalert/store/alert_rule.go:69]: not for this change, but we should make this sort of IN clause easier to construct in a utility in the sqlstore or related package.

File: pkg/services/ngalert/store/alert_rule.go [hunk 3]
```diff
@@ -64,20 +64,30 @@ func getAlertRuleByUID(sess *sqlstore.DBSession, alertRuleUID string, orgID int6
...
+
+		args := make([]interface{}, 0, len(ruleUID)+2)
+		args = append(args, "DELETE FROM alert_rule WHERE org_id = ? AND uid IN "+params)
+		args = append(args, orgID)
+		for _, arg := range ruleUID {
+			args = append(args, arg)
+		}
+
+		_, err := sess.Exec(args...)
```
Comment (joeblubaugh) [inline pkg/services/ngalert/store/alert_rule.go:78]: Also not for this pr, but I'm surprised we don't differentiate the query string itself from the query arguments ...

**Review by joeblubaugh — APPROVED** [review]

File: pkg/services/ngalert/store/alert_rule.go [hunk 4]
```diff
@@ -64,20 +64,30 @@ func getAlertRuleByUID(sess *sqlstore.DBSession, alertRuleUID string, orgID int6
 }
 
 // DeleteAlertRuleByUID is a handler for deleting an alert rule.
-func (st DBstore) DeleteAlertRuleByUID(ctx context.Context, orgID int64, ruleUID string) error {
+func (st DBstore) DeleteAlertRulesByUID(ctx context.Context, orgID int64, ruleUID ...string) error {
 	return st.SQLStore.WithTransactionalDbSession(ctx, func(sess *sqlstore.DBSession) error {
-		_, err := sess.Exec("DELETE FROM alert_rule WHERE org_id = ? AND uid = ?", orgID, ruleUID)
```
Comment (gotjosh) [inline pkg/services/ngalert/store/alert_rule.go:69]: As @joeblubaugh said, could we perhaps take the opportunity to improve the way we handle arguments for these IN clauses?

I think something like the below should work (you can choose to use or not use the "affected rows") - but you get the gist of it.

```golang
rows, err := sess.Table("alert_rule").Where("org_id = ?", orgID).In("uid", ruleUID).Delete(ngmodels.AlertRule{})
if err != nil {
	return err
}
st.Logger.Debug("deleted alert rules", "count", rows)

rows, err = sess.Table("alert_rule_version").Where("rule_org_id = ?", orgID).In("rule_uid", ruleUID).Delete(ngmodels.AlertRuleVersion{})
if err != nil {
	return err
}
st.Logger.Debug("deleted alert rule versions", "count", rows)

rows, err = sess.Table("alert_instance").Where("rule_org_id = ?", orgID).In("rule_uid", ruleUID).Delete(ngmodels.AlertInstance{})
if err != nil {
	return err
}
st.Logger.Debug("deleted alert instances", "count", rows)
```

**Review by gotjosh — COMMENTED** [review]
LGTM on avoiding the N+1 query, but please see my comment.

File: pkg/services/ngalert/store/alert_rule.go (hunk 4 above)
Comment (yuri-tceretian) [reply pkg/services/ngalert/store/alert_rule.go:69]: Oh, that's nice xorm-foo. I am personally inclined in using xorm methods but could not find that technique in docs

File: pkg/services/ngalert/schedule/schedule_test.go (hunk 1 above)
Comment (gotjosh) [reply pkg/services/ngalert/schedule/schedule_test.go:13]: It is not enforced by `gofmt` - but it is the preferred style.

**Review by gotjosh — APPROVED** [review]
LGTM

**Pull request merged**